README.md

```diff
@@ -1,56 +1,72 @@
 Currently, the project works with the following libraries:
 * [gorilla/websocket](https://github.com/gorilla/websocket): a websocket package for golang. We chose to use this 
 package because it offers a good API, and it is more complete than the websocket package offered by the standard library.
 * [boltdb](https://github.com/boltdb/bolt): a low-level key-value DB management system. It works by creating Buckets, 
 that we can populate with key-value pairs.
 
-## WebSocket Protocol
-The actual implementation (4 files) currently works as following :
-* the `hub.go` maintains a list of opened connections and "routes" the messages. We will change this so there is no more routing to do in the next update.
-* the `connection.go` is the file that properly handles the websocket connection. When it receives a message from the client, it passes 
-it to the hub.
-* the `index.html` acts like the client that prints messages received from the server and that can send messages to the server.
-* the `homePage.go` just serves the `index.html` file on http requests
-### tweaking the server's properties 
+### network testing 
 If you want to test the websocket protocol the server running on a different machine than the client, you have to set the
 server's IP address in `index.html`  at line 36 (instead of `localhost`). You'll have to put your computer's IP address
 before the ":8080" of `main.go` at line 29.
 
-## Database Structure
-As we are using a very basic DBMS enabling only key-values storage. There are currently 3 different databases :
-1. for the hub, containing a list of channels and their subscribers
-2. one for the witness (described below)
-3. one for the organizer (same structure as witness's one)
+### Database Structure
+Currently, there are 2 databases created/opened when we run the backend: orgDatabase.db/witDatabase.db and sub.db.
+
+#### Sub.db
+This database exists only to handle publisher and subscribers of channels. Currently, we only manage subscribers. There is a bucket called
+"sub", in which we store the (key, value) pairs as following : each key is the ID of one channel, and the value is an array of connections
+that subscribed to this channel. When someone subscribes or unsubscribes, we simply edit this array.
+
+#### witDatabase.db / OrgDatabase.db
+This file has the same structure, whether it is witness or organizer. It has 2 types of bucket :
+* The first one is called `channels` and for each channel it contains a (key, value) pair, where the key is the channel's ID
+and the value is a string containing the "current state" of the channel. There is no history of messages stored here, only the channel's name, organizer, list of witnesses, etc...
+* The second type of bucket is the bucket containing the history of the channel. There is one such bucket per channel, and 
+it has the following properties :
+    1. The bucket name is the ID of the channel
+    2. The (key, values) pairs are (ID, content) of messages sent on the channel.
```

Comment:
Is this a list of (key, value) pairs or a dictionary, indexed by key ? 
How can you (efficiently) respond to a simple query such as "give me all the messages between time X and time Y"?

Reply:
If I am not mistaken, it is indexed by key (with a B+Tree index). I also just found out there is a fork of boltDB, kept up to date and actively maintained, so I might consider switching to it.

Anyway, we currently have no efficient way to execute this query. We could build an index ourselves, by having a second bucket, containing key-values pairs with keys being timestamps and values the message sent/received at this precise time. This would double the storage required, but we could execute the query in O(logN + M) time, instead of O(N), where N is the number of entries and M the number of items corresponding to the search. We can do this quite easily, but we need to be sure it is "worth it".

To answer your first question, we use a simple key-value storage, and not a more complex relational storage, because @bford told us so in the beginning of the project. It's true we have not tried to doubt this, though as the overall goal is to create a mobile App, I think the biggest advantage is that it's lighter and so needs less ressource.

Though, if you (and Brian) think it's worth it to reconsider using a relational database, I'll gladly do it. Maybe we should discuss that in the next meeting ?

Reply:
Yes, let's discuss this. Sqlite (for instance) could be just as lightweight while offering more thorough functionality. 
Changing storage backend might not be the highest priority task either, and we could consider it out of scope, while taking note that it might be worth doing later on :)

Reply:
Even if I have no idea of the performance impact, I don't think it makes sense to change this on week 14. We'll wirte it down as an improvement suggestion for later, but it's most probably not going to be implemented before the end of the project.